### PR TITLE
pytz upgraded to 2024.2

### PR DESCRIPTION
**Pull request by vutrobinnj**
## Ticket #7777

Vitaliy found that newer `croniter` versions pulled forward the version of `pytz`, which conflicts with `pyopenfec`'s preferred pinned version (`2019.3`):
https://github.com/NationalJournal/pyopenfec/blob/31cd391da47294795e111541c094062a8b7aac73/requirements.txt#L1-L3

It's possible that `pyopenfec` would work fine if we removed/changed the pinned version -- we have our own `pyopenfec` repo so we can edit the requirements. That could let us upgrade `croniter` to a newer version. After this batch of updates, Vitaliy may circle back to try that.

## Testing
- run `pip install pytz==2024.2`
- on local run `pip install git+https://github.com/NationalJournal/pyopenfec@7777-upgrade-pytz-lib`
- run `python manage.py quarterly_finance_report --year 2024 --quarter Q3 --body H` confirm that it runs without errors.
- run `pip install pytz==2019.3` return back to original version.

### CODE REVIEW

**Comment by cdornfeldnj** [comment]
Noting that we may need to coordinate the merge of the version bump in this PR with the corresponding `pytz` version bump in `nj-cms`. Please **do not merge** without discussion first.

Also:
- [x] Please bump the library version number

We're not currently referencing this library by a specific version from the `requirements.txt` in `nj-cms`, so whatever is merged into `master` here is getting pulled in. But later on, we could begin pinning a specific version with this syntax:
```
package-two @ git+https://github.com/owner/repo@releases/tag/v3.7.1
```

For testing before review/merge, we should be able to specify a specific branch locally (temporarily only) as follows:
```
package-two @ git+https://github.com/owner/repo@mybranch
```

Reference: https://stackoverflow.com/a/35998253

---

After we merge this PR, reminder to:
- [x] Create a new tag in GitHub for `0.2.8`
- [ ] Update the upstream `requirements.txt` to reference the new version tag for `pyopenfec`
- [ ] Update the upstream `requirements.txt` to reference the same new version of `pytz`

**Comment by cdornfeldnj** [comment]
@vutrobinnj please double-check any tests now that the main `nj-cms` requirements file has pinned versions. If tests are passing OK, we can advance this for code review.

**Comment by vutrobinnj** [comment]
@cdornfeldnj 

> @vutrobinnj please double-check any tests now that the main `nj-cms` requirements file has pinned versions. If tests are passing OK, we can advance this for code review.

Update to requirements.txt to reference tagged versions has been pushed to master last week.
https://github.com/NationalJournal/nj-cms/pull/7787

It should be ok to start review on this PR.

**Comment by cdornfeldnj** [comment]
@vutrobinnj just double-checking:

- How does the `quarterly_finance_report` command know to pick up `pytz==2024.2` for testing this change -- doesn't it follow the hard-coded version in our `requirements.txt` now? Do testers need to temporarily modify `requirements.txt` in `nj-cms`?
https://github.com/NationalJournal/nj-cms/blob/staging/NJ/requirements.txt#L161-L163
```
git+https://github.com/NationalJournal/pyopenfec@0.2.7
```

- Any steps needed to return to the usual `pytz` version in a local environment after the other testing steps?

**Comment by vutrobinnj** [comment]
@cdornfeldnj 
Step 1 of testing will make sure that `pytz==2024.2`

> @vutrobinnj just double-checking:
> 
> * How does the `quarterly_finance_report` command know to pick up `pytz==2024.2` for testing this change -- doesn't it follow the hard-coded version in our `requirements.txt` now? Do testers need to temporarily modify `requirements.txt` in `nj-cms`?
>   https://github.com/NationalJournal/nj-cms/blob/staging/NJ/requirements.txt#L161-L163
> 
> ```
> git+https://github.com/NationalJournal/pyopenfec@0.2.7
> ```
> 
> * Any steps needed to return to the usual `pytz` version in a local environment after the other testing steps?

> ## Testing
- run `pip install pytz==2024.2`
- on local run `pip install git+https://github.com/NationalJournal/pyopenfec@7777-upgrade-pytz-lib`
- run `python manage.py quarterly_finance_report --year 2024 --quarter Q3 --body H` confirm that it runs without errors.

**Comment by cdornfeldnj** [comment]
@vutrobinnj OK, so at the end of the testing instructions, should testers do similar `pip install` steps with the "old" versions to get their local environments back in sync with `nj-cms@staging` before we merge/tag this PR?

**Comment by vutrobinnj** [comment]
@cdornfeldnj 
> @vutrobinnj OK, so at the end of the testing instructions, should testers do similar `pip install` steps with the "old" versions to get their local environments back in sync with `nj-cms@staging` before we merge/tag this PR?

They could, ( I updated the testing section), however it does not change anything.
Once this PR is merged to pyopenfec master, We need to bump up pyopenfec and pytz versions in njcms.

**Review by alittler-nj — APPROVED** [review]

**Comment by cdornfeldnj** [comment]
Follow-up needed per Jenkins:

```
[2024-12-03T14:47:19.033Z] The conflict is caused by:
[2024-12-03T14:47:19.033Z]     The user requested pytz==2024.2
[2024-12-03T14:47:19.033Z]     pandas 1.1.5 depends on pytz>=2017.2
[2024-12-03T14:47:19.033Z]     fecfile 0.4.5 depends on pytz>=2018.4
[2024-12-03T14:47:19.033Z]     celery 4.3.0 depends on pytz>dev
[2024-12-03T14:47:19.033Z]     flower 0.9.3 depends on pytz
[2024-12-03T14:47:19.033Z]     croniter 5.0.1 depends on pytz>2021.1
[2024-12-03T14:47:19.033Z]     django 1.11.24 depends on pytz
[2024-12-03T14:47:19.033Z]     pyopenfec 0.2.8 depends on pytz<2020.0 and >=2019.3
```

https://github.com/NationalJournal/pyopenfec/blob/2fdda73424abf3bef25343a1cd2ede82690f3150/pyproject.toml#L32-L35

- [ ] Change to:
```
pytz = "^2024.2"
```

`^` syntax is ">= this version but less than the next major version"

Poetry version reference: https://python-poetry.org/docs/dependency-specification/

Then:
- [ ] Delete `0.2.8` tag from this repo
- [ ] Recreate `0.2.8` tag after merging this follow-up change
- [ ] For any environments (probably just Vitaliy's local dev) where the old 0.2.8 was pip-installed, clean it out/reinstall

**Pull request merged**
